### PR TITLE
[Refactoring] Remove unfinished imap protocol migration

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: server/protocols/protocols-imap4/src/main/java/org/apache/james/imapserver/netty/IMAPMDCContext.java [hunk 1]
```diff
@@ -59,14 +62,17 @@ private static String retrieveHost(ChannelHandlerContext ctx) {
...
-            .filter(object -> object instanceof IMAPSession)
-            .map(object -> (IMAPSession) object)
-            .map(imapSession -> MDCBuilder.create()
-                .addContext(MDCBuilder.SESSION_ID, imapSession.getSessionID())
-                .addContext(MDCBuilder.USER, imapSession.getUser())
-                .addContext(from(Optional.ofNullable(imapSession.getSelected()))))
-            .orElse(MDCBuilder.create());
+        if (o instanceof ImapSession) {
+            ImapSession imapSession = (ImapSession) o;
+            MailboxSession mailboxSession = (MailboxSession) imapSession.getAttribute(ImapSessionUtils.MAILBOX_SESSION_ATTRIBUTE_SESSION_KEY);
```
Comment (rouazana) [inline server/protocols/protocols-imap4/src/main/java/org/apache/james/imapserver/netty/IMAPMDCContext.java]: you don't check anymore for nullity?

Comment (chibenwa) [reply server/protocols/protocols-imap4/src/main/java/org/apache/james/imapserver/netty/IMAPMDCContext.java]: It build is :green_apple:  this means no NPE this means not necessary.

Note null checks line 70. And I am not aware of casts being forbidden for null...

Comment (rouazana) [reply server/protocols/protocols-imap4/src/main/java/org/apache/james/imapserver/netty/IMAPMDCContext.java]: > It build is green_apple this means no NPE this means not necessary.

I agree in general, but I prefer to be very cautious when code is related to logging. In general you log when something not really expected is coming...

> Note null checks line 70. And I am not aware of casts being forbidden for null...

Of course, that's why I was commenting on line 67 where you do imapSession.getAttribute

Comment (chibenwa) [reply server/protocols/protocols-imap4/src/main/java/org/apache/james/imapserver/netty/IMAPMDCContext.java]: instanceof protects against o being null, nullchecks on o is thus not needed.

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**